libraries/aws_iam_user.rb

```diff
@@ -15,29 +15,32 @@ class AwsIamUser < Inspec.resource(1)
   def initialize(
     opts,
     aws_user_provider = AwsIam::UserProvider.new,
+    aws_user_details_provider_factory = AwsIam::UserDetailsProviderFactory.new,
     access_key_factory = AwsIamAccessKeyFactory.new
   )
     @user = opts[:user]
     @user = aws_user_provider.user(opts[:name]) if @user.nil?
+    @aws_user_details_provider = aws_user_details_provider_factory.create(@user)
```

Comment:
I think we create exactly one details provider per user resource, in which case it might be overkill to pass in a factory... you can just pass in an instance directly.

(I usually provide a factory abstraction only when the, e.g., aws_iam_user, needs to create N instances in response to method calls after instantiation.)

Reply:
Maybe `Factory` was a poor name for this abstraction, because you're correct, there is only ever one instance of the `UserDetailsProvider` here.

The idea we were having when we separated out the `Details` from the `UserProvider`, was that the `Details` should somehow be initialized with the underlying user. If we inject an instance of the `UserDetailsProvider` here, it needs to have already been created with a specific user.

If you take a look at some our earlier commits https://github.com/chef/inspec-aws/pull/89/commits/220acacbc7de17a8ca2679a94974ba50673b6b18, we injected a `UserDetailsProvider` and then associated it to a specific user in the ctor (like a setter), but decided against this approach as then the details provider  would be mutable, and it made the tests too weird.